_posts/2025-05-20-highlights-otp-28.md

```diff
@@ -0,0 +1,662 @@
+---
+layout: post
+title: Erlang/OTP 28 Highlights
+tags: erlang otp 28 release
+author: Isabell Huang
+---
+
+Erlang/OTP 28 is finally here. This blog post will introduce the new
+features that we are most excited about.
+
+A list of all changes is found in [Erlang/OTP 28 Readme](https://erlang.org/patches/OTP-28.0).
+Or, as always, look at the release notes of the application you are interested in.
+For instance:
+[Erlang/OTP 28 - Erts Release Notes - Version 15.0](https://www.erlang.org/doc/apps/erts/notes.html#erts-16.0).
+
+This year's highlights mentioned in this blog post are:
+
+* [Priority Messages](#priority-messages)
+* [Improvements of Comprehensions](#improvements-of-comprehensions)
+* [Smarter Error Suggestions](#smarter-error-suggestions)
+* [Improvements to the Shell](#improvements-to-the-shell)
+* [New `hibernate/0`](#new-hibernate0)
+* [Warnings for Use of Old-style Catch](#warnings-for-use-of-old-style-catch)
+* [PCRE2](#pcre2)
+* [Based Floating Point Literals](#based-floating-point-literals)
+* [Nominal Types](#nominal-types)
+
+
+# Priority Messages
+
+Sometimes, it is important for urgent messages to _skip the queue_ and
+be read by the receiving process as soon as possible. Erlang/OTP 28 introduces
+priority messages, an opt-in mechanism that allows the receiving process
+to let certain messages get priority status.
+
+By default, all messages are inserted to the end of the message queue of
+a process. This can become cumbersome when the queue is long. An urgent
+message may need to be read as soon as possible.
```

Comment:
```suggestion
message may need to be read as soon as possible.

For example, the current message overload protection mechanism for `logger` polls its message
queue length in order to know when it should start shedding messages. It would have benefitted
from using the [`long_message_queue`](https://www.erlang.org/doc/apps/erts/erlang.html#alias/1)
monitoring functionality introduced in Erlang/OTP 27, but the only way to get information like that is
via a message which would have been inserted at the back of the very large queue.
```